actions/main/src/gradleinttest.sh

```diff
@@ -51,12 +51,17 @@ gradleinttest () {
       echo "Running integration tests against ${IT_JADE_API_URL}"
       cleaniampolicy
     fi
+    if [[ "${test_to_run}" == "check" ]] && [[ "${SONAR_TOKEN}" != "" ]]; then
```

Comment:
The code use `-n` elsewhere so why not use it here too?
```suggestion
    if [[ "${test_to_run}" == "check" ]] && [[ -n "${SONAR_TOKEN}" ]]; then
```

Although if you use `[[` you may as well put the `&&` inside it. This should work
```suggestion
    if [[ $test_to_run == check && -n $SONAR_TOKEN ]]; then
```
- inside `[[`, you don't need to quote `$` expansions to prevent globbing
- bash doesn't have strings, so quoting `check` has no effect
- `[[` (unlike `[`) supports `&&` and `||` directly so they don't need to be separated

Reply:
Ok, great! This is much cleaner!